### PR TITLE
Increase Course Description field length

### LINKED ISSUE #383: Expand the Description field in the Course Object to support > 255 characters
The current field is limited to 255 characters. Most of the course descriptions for educational institutions are > 255 characters. TES's description field is 7000 characters long, but this seems excessively long. Review of several course catalogs from universities around the country suggests almost all course descriptions are < 1500 characters. 

Make the Description field in the Course Object 2048 characters to account for the vast majority of course descriptions available. > 2048 character descriptions should be truncated or a custom field can be created to support it.

### PR DESCRIPTION
**Pull request by davidjray**
# Critical Changes

# Changes
- Added Extended Description, a new field to accommodate course descriptions up to 131,072 characters.
- In the Course page layout, replaced the existing course Description field with the new Extended Description field.
- In HEDA Settings, added a Courses tab with a new batch process that lets you easily copy existing values from a course's Description to Extended Description.

# Issues Closed
Fixes #383

### CODE REVIEW

**Comment by davidjray** [comment]
This PR includes additional metadata, mostly required flags, (including the required flag). This is to include the metadata changes retrieved from running `retrieve_src` command in CumulusCi 2.

**Comment by davidjray** [comment]
Screenshot of the new section in HEDA Settings to run the copy procedure.
screen shot 2017-03-21 at 2 10 26 pm

File: src/aura/STG_CMP_CourseConnections/STG_CMP_CourseConnections.cmp [hunk 1]
```diff
@@ -172,4 +172,33 @@
...
+          <h2>
+              <span class="slds-text-heading--small">Courses: Description Data Migration</span>
+          </h2>
+        </div>
+      </header>
+    </div>
+    <div class="slds-card__body">
+      <div class="slds-p-around--medium" style="background-color: #ffffff;">
+        <!-- Card Body -->
+        In HEDA 1.30 a new extended field was added to accommodate longer Course descriptions.<br />
```
Comment (davidjray) [inline src/aura/STG_CMP_CourseConnections/STG_CMP_CourseConnections.cmp]: @lianeyee When you have a free minute, would you look over these instructions (there is also a screenshot I posted) to make sure they are clear.

Comment (lianeyee) [inline src/aura/STG_CMP_CourseConnections/STG_CMP_CourseConnections.cmp]: @davidjray Thanks for the heads up! Reviewing now.

File: src/layouts/Course__c-HEDA Course Layout.layout [hunk 2]
```diff
@@ -16,7 +16,7 @@
             </layoutItems>
             <layoutItems>
                 <behavior>Edit</behavior>
-                <field>Description__c</field>
+                <field>Extended_Description__c</field>
```
Comment (lianeyee) [inline src/layouts/Course__c-HEDA Course Layout.layout]: Good suggestion, Carlos. I agree we should change if we can.

Comment (davidjray) [inline src/layouts/Course__c-HEDA Course Layout.layout]: Actually yes, I was instructed to have the old field removed from the page layout by @wongmike. This is the reason behind the batch process, so admins of existing Orgs can move the data over after updating their existing layout.

File: src/classes/COUR_DescriptionCopy_BATCH.cls [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    /*******************************************************************************************************
+    * @description start Method for the Database.Batchable interface
+    * @param bc the BatchableContext
+    * @return database.Querylocator
+    ********************************************************************************************************/
+	global Database.QueryLocator start(Database.BatchableContext BC) {
+
+		// Get all courses with a description and NOT an extended description
+		String query = 'SELECT Id, Name, Description__c, Extended_Description__c FROM Course__c WHERE Description__c != null';
```
Comment (lianeyee) [inline src/classes/COUR_DescriptionCopy_BATCH.cls]: Re: #5 above, maybe "Courses and Connections"?

File: src/classes/COUR_DescriptionCopy_BATCH.cls [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+   	global void execute(Database.BatchableContext BC, List<sObject> scope) {
+		List<Course__c> courses = (List<Course__c> ) scope; //Cast list of courses
+        
+        if(courses.size()>0) {
+	        for(Course__c c : courses) {
+
+	        	// Check extended description is blank, since we cannot query for this field in our where clause
+	        	if(c.Description__c != null && c.Extended_Description__c == null) {
```
Comment (davidjray) [inline src/classes/COUR_DescriptionCopy_BATCH.cls]: Updated!

File: src/classes/COUR_DescriptionCopy_BATCH.cls [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+							FROM AsyncApexJob WHERE Id =:BC.getJobId()];
+
+		String emailBody = 'The batch Apex job processed ' + a.TotalJobItems +
+		' batches with '+ a.NumberOfErrors + ' failures.';
+
+		// Send an email to the Apex job's submitter notifying of job completion.
+		Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+		String[] toAddresses = new String[] {a.CreatedBy.Email};
+		mail.setToAddresses(toAddresses);
+		mail.setSubject('Course Description Copy Job Completed: ' + a.Status);
```
Comment (davidjray) [inline src/classes/COUR_DescriptionCopy_BATCH.cls]: @ceiroa Would you suggest using a parameterized version of the Label by taking advantage of `String.format()`? I notice we are not using any parameterized Labels now.

**Pull request merged**